libs/transactions/units.ts

```diff
@@ -0,0 +1,57 @@
+import Web3 from "web3";
+
+function isString(s: any) {
+  return typeof s === "string" || s instanceof String;
+}
+
+export function toBaseUnit(value: any, decimals: number, web3: Web3) {
+  if (!isString(value)) {
+    throw new Error("Pass strings to prevent floating point precision issues.");
+  }
+  if (decimals === 0) {
+    return value;
+  }
+
+  const ten = web3.utils.toBN(10);
+  const base = ten.pow(web3.utils.toBN(decimals));
+
+  // Is it negative?
+  const negative = value.substring(0, 1) === "-";
+  if (negative) {
+    throw new Error(`Invalid  value cannot be converted negative`);
+  }
+
+  if (value === ".") {
+    throw new Error(
+      `Invalid  value ${value} cannot be converted to${+`  base unit with  $ { decimals }  decimals .`}`,
+    );
+  }
+
+  // Split it into a whole and fractional part
+  const comps = value.split(".");
+  if (comps.length > 2) {
+    throw new Error("Too many decimal points");
+  }
+
+  let whole = comps[0];
+  let fraction = comps[1];
```

Comment:
I think an error could be raised when accessing comps[1] if the `value` parameter doesn't have a `.`.

Reply:
Of course, I will also consider this as part of issue #28